src/containers/ConnectorManagement.jsx

```diff
@@ -171,19 +154,55 @@ export default class ConnectorManagement extends Component {
         }
       })
       .catch(error => { // eslint-disable-line
-        this.setState({ submitting: false })
-        console.error(error)
         Notifier.error(t('account config error'))
         this.gotoParent()
         throw error
       })
   }
 
-  connectAccountOAuth (values) {
-    return this._updateAccount(0, values)
-      .then(() => {
-        window.open(prepareConnectURL(this.state.connector), 'width=800,height=800')
+  terminateOAuth (storageEvent) {
+    const { t } = this.context
+    if (storageEvent.key !== 'oauth_terminating') return // ignore other keys
+    // get account id from localStorage event and remove the listener
+    const accountID = JSON.parse(storageEvent.newValue).key
```

Comment:
You should check before that storageEvent is really an object and handle JSON.parse exception. There are some cases where it is null at least on firefox.